### ISSUE TITLE
On instantiation of the SQLClient class, models stored is empty

### ISSUE THREAD

**Issue by ohlookemus** [dgilland/sqlservice#12]
This started happening after I upgraded to 0.10.0 to fix the duplicate Model name issue I reported earlier. We discovered that we started getting errors such as 

`Type of value given to save() method is not a valid SQLALchemy declarative class. Item with index 0 and with value "`

Upon further debugging by looking at the instance of SQLClient via .models.values() we discovered an empty dictionary. I saw in the 0.10.0 change log that there was some kind of code change for this. I'm not sure if it's related to how we instantiate the base class but I've posted a bit of our code to help out.

database.py
```
from sqlservice import SQLClient
from orm import Model
from config import BaseConfig

config = {
    'SQL_DATABASE_URI': 'postgresql://{0}:{1}@{2}:{3}/{4}'.format(
        BaseConfig.DATABASE['DB_USER'], BaseConfig.DATABASE['DB_PASS'],
        BaseConfig.DATABASE['DB_SERVICE'], BaseConfig.DATABASE['DB_PORT'],
        BaseConfig.DATABASE['DB_NAME'])
}

db = SQLClient(config, model_class=Model)
```

orm.Model
```
import datetime

import sqlalchemy as sa

from sqlservice import declarative_base, ModelBase

from sqlalchemy import MetaData
from sqlalchemy.ext.declarative import declared_attr


meta = MetaData(naming_convention={
    "ix": 'ix_%(column_0_label)s',
    "uq": "uk_%(table_name)s_%(column_0_name)s",
    "ck": "ck_%(constraint_name)s",
    "fk": "fk_%(column_0_name)s_%(referred_table_name)s",
    "pk": "pk_%(table_name)s"
}, schema='public')

class MyCustomBase(ModelBase):
    @declared_attr
    def created_by(cls):
        return sa.Column(sa.SmallInteger, sa.ForeignKey('public.user.user_id'))

    @declared_attr
    def created_ts(cls):
        return sa.Column(sa.DateTime(timezone=False), nullable=False,
                         default=datetime.datetime.utcnow,
                         server_default=sa.func.now())

    @declared_attr
    def modified_by(cls):
        return sa.Column(sa.SmallInteger, sa.ForeignKey('public.user.user_id'))

    @declared_attr
    def modified_ts(cls):
        return sa.Column(sa.DateTime(timezone=False),
                         onupdate=datetime.datetime.utcnow,
                         server_onupdate=sa.func.now())


Model = declarative_base(MyCustomBase, metadata=meta)
```

Subsequent model files then import Model from orm.Model like so
```
import sqlalchemy as sa
from sqlalchemy.dialects.postgresql.json import JSON

from orm import Model

class User(Model):
    __tablename__ = 'user'

    user_id = sa.Column(sa.Integer, primary_key=True)
```

When I downgrade to 0.9.1, everything works as expected.

**Comment by dgilland** [comment]
Can you provide a failing example? Whenever I plugged the code you provided into a script, I'm not seeing any issues.

```python
# issue12.py

import datetime

import sqlalchemy as sa

from sqlservice import declarative_base, ModelBase, SQLClient

from sqlalchemy import MetaData
from sqlalchemy.ext.declarative import declared_attr


meta = MetaData(naming_convention={
    "ix": 'ix_%(column_0_label)s',
    "uq": "uk_%(table_name)s_%(column_0_name)s",
    "ck": "ck_%(constraint_name)s",
    "fk": "fk_%(column_0_name)s_%(referred_table_name)s",
    "pk": "pk_%(table_name)s"
}, schema='public')


class MyCustomBase(ModelBase):
    @declared_attr
    def created_by(cls):
        return sa.Column(sa.SmallInteger, sa.ForeignKey('public.user.user_id'))

    @declared_attr
    def created_ts(cls):
        return sa.Column(sa.DateTime(timezone=False), nullable=False,
                         default=datetime.datetime.utcnow,
                         server_default=sa.func.now())

    @declared_attr
    def modified_by(cls):
        return sa.Column(sa.SmallInteger, sa.ForeignKey('public.user.user_id'))

    @declared_attr
    def modified_ts(cls):
        return sa.Column(sa.DateTime(timezone=False),
                         onupdate=datetime.datetime.utcnow,
                         server_onupdate=sa.func.now())


Model = declarative_base(MyCustomBase, metadata=meta)


class User(Model):
    __tablename__ = 'user'

    user_id = sa.Column(sa.Integer, primary_key=True)



config = {
    'SQL_DATABASE_URI': 'postgresql://***:***@localhost:5432/***'
}

db = SQLClient(config, model_class=Model)
db.create_all()

print('models:', db.models)

db.close()
db.drop_all()
```

which outputs:

```
$ python issue12.py
models: {'User': <class '__main__.User'>}
```

However, if I move the creation of `SQLClient` to before the creation of the `User` class, then I do see an empty `dict` for `db.models`. So in your case, maybe you have created `db` before your models have been imported? You can verify this by inspecting `Model` right before you create `db`:

```python
print(list(Model._decl_class_registry.items()))
```

That's what `SQLClient` uses to populate `db.models`. So if there are no ORM models attached to `Model`, when you do something like `db.save(User())`, you'll get the error message you mentioned. This is actually an oversight on my part. To support model imports that happen after `db` is created, I update `db.models` whenever one tries to do `db.User` and `User` doesn't exist in `db.models`. So `db.User.save()` would work but not `db.save(User())` since I'm not trying to update `db.models` there (but I should be).

So if you can confirm that this is what's happening, then I can push out a fix for it. Otherwise, if it's some other issues, a failing example from you would help me debug further.

**Comment by ohlookemus** [comment]
It's definitely an import order error as I have SQLClient loaded first when my Flask application starts because some of the models are loaded via blueprints but the db client is instantiated first.

**Comment by dgilland** [comment]
I've released `v0.11.0` that addresses the late model imports: https://pypi.python.org/pypi/sqlservice/0.11.0

Let me know if that fixes your issue or not.

**Comment by ohlookemus** [comment]
Looks good!
